ext/spl/spl_directory.c

```diff
@@ -478,8 +478,7 @@ static spl_filesystem_object *spl_filesystem_object_create_info(spl_filesystem_o
 	zval arg1;
 
 	ZEND_ASSERT(file_path && ZSTR_LEN(file_path) > 0);
-
-	ce = ce ? ce : source->info_class;
+	ZEND_ASSERT(ce != NULL);
```

Comment:
Why can `ce` not be NULL? Nothing prevents users from passing NULL to the function `getPathInfo` AFAICT?

Reply:
I'll check and add a test if it can